server/http/handlers/service.go

```diff
@@ -120,20 +155,25 @@ func (s *service) Query(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	unmarshalStart := s.nowFn()
 	var q query.RawQuery
-	if err = json.Unmarshal(data, &q); err != nil {
+	err = json.Unmarshal(data, &q)
+	s.metrics.unmarshal.ReportSuccessOrError(err, s.nowFn().Sub(unmarshalStart))
+	if err != nil {
 		err = fmt.Errorf("unable to unmarshal request into a raw query: %v", err)
 		writeErrorResponse(w, err)
 		return
 	}
 
+	parseStart := s.nowFn()
 	parseOpts := query.ParseOptions{
 		FieldPathSeparator:    s.dbOpts.FieldPathSeparator(),
 		FieldHashFn:           s.dbOpts.FieldHashFn(),
 		TimestampFieldPath:    s.dbOpts.TimestampFieldPath(),
 		RawDocSourceFieldPath: s.dbOpts.RawDocSourceFieldPath(),
 	}
 	pq, err := q.Parse(parseOpts)
+	s.metrics.parse.ReportSuccessOrError(err, s.nowFn().Sub(parseStart))
```

Comment:
Take it or leave it, but I doubt this is material wrt the overall query latency.